airflow/www/views.py

```diff
@@ -972,42 +972,58 @@ def delete(self):
         # Upon success return to origin.
         return redirect(origin)
 
-    @expose('/trigger', methods=['POST'])
+    @expose('/trigger', methods=['POST', 'GET'])
     @has_dag_access(can_dag_edit=True)
     @has_access
     @action_logging
     @provide_session
     def trigger(self, session=None):
+
         dag_id = request.values.get('dag_id')
         origin = request.values.get('origin') or url_for('Airflow.index')
-        dag = session.query(models.DagModel).filter(models.DagModel.dag_id == dag_id).first()
-        if not dag:
-            flash("Cannot find dag {}".format(dag_id))
-            return redirect(origin)
 
-        execution_date = timezone.utcnow()
-        run_id = "manual__{0}".format(execution_date.isoformat())
+        if request.method == 'GET':
+            return self.render_template(
+                'airflow/trigger.html',
+                dag_id=dag_id,
+                origin=origin
+            )
+        else:
```

Comment:
Since the GET path returns, you can drop the `else:` (and not have to touch the rest of this method (making the diff easier to review.